ql/src/java/org/apache/hadoop/hive/ql/exec/Utilities.java

```diff
@@ -4587,4 +4594,20 @@ private static void tryToLogClassPath(String prefix, ClassLoader loader, Logger
           loader == null ? "null" : loader.getClass().getSimpleName());
     }
   }
+
+  public static String encodeColumnNames(List<String> colNames) throws SemanticException {
+    try {
+      return JSON_MAPPER.writeValueAsString(colNames);
+    } catch (IOException e) {
+      throw new SemanticException(e);
+    }
+  }
+
+  public static List<String> decodeColumnNames(String colNamesStr) throws SemanticException {
+    try {
+      return JSON_MAPPER.readValue(colNamesStr, List.class);
```

Comment:
JSON_MAPPER is just a Jackson `ObjectMapper` object that can be used to convert JSON into objects and vice versa. In particular, `colNamesStr` here is the JSON value (in String form) and we pass the `List.class` because we want to generate an instance of that object.